cmd/copy.go

```diff
@@ -1480,17 +1480,31 @@ func init() {
 		Example:    copyCmdExample,
 		Args: func(cmd *cobra.Command, args []string) error {
 			if len(args) == 1 { // redirection
-				if stdinPipeIn, err := isStdinPipeIn(); stdinPipeIn == true {
-					raw.src = pipeLocation
-					raw.dst = args[0]
+				var userFromTo common.FromTo
+				if raw.fromTo == "" {
+					userFromTo = common.EFromTo.Unknown()
 				} else {
+					err := userFromTo.Parse(raw.fromTo)
 					if err != nil {
-						return fmt.Errorf("fatal: failed to read from Stdin due to error: %s", err)
-					} else {
-						raw.src = args[0]
-						raw.dst = pipeLocation
+						return fmt.Errorf("fatal: incorrect from-to argument passed: %s", raw.fromTo)
 					}
 				}
+
+				stdinPipeIn, err := isStdinPipeIn()
+				if err != nil {
+					return fmt.Errorf("fatal: failed to read from Stdin due to error: %s", err)
+				}
+
+				if (userFromTo == common.EFromTo.PipeBlob() || userFromTo == common.EFromTo.Unknown()) && stdinPipeIn == true {
```

Comment:
>if (userFromTo == common.EFromTo.PipeBlob() || userFromTo == common.EFromTo.Unknown()) && stdinPipeIn == true { [](start = 4, length = 111)

I don't believe this covers the "if both pipes are in with no fromto, error out" scenario.

Reply:
Although, since it doesn't look like we're actually checking for stdout as a pipe... I don't think we can do that.

---
In reply to: [494506908](https://github.com/Azure/azure-storage-azcopy/pull/1184#discussion_r494506908) [](ancestors = 494506908)

Reply:
I think the best option is to just error out if we don't know the fromto in this case, and publicly document BlobPipe and PipeBlob.

---
In reply to: [494507871](https://github.com/Azure/azure-storage-azcopy/pull/1184#discussion_r494507871) [](ancestors = 494507871,494506908)

Reply:
though that might be a breaking change...

---
In reply to: [494508421](https://github.com/Azure/azure-storage-azcopy/pull/1184#discussion_r494508421) [](ancestors = 494508421,494507871,494506908)

Reply:
Enforcing `from-to` flag. 
Validated cases
-  Upload/Download when no `from-to` flag provided => `fatal: from-to flag required`
-  Upload/Download when invalid `from-to` flag provided (BlobTrash, LocalBlob) => `fatal: invalid from-to flag argument passed: BlobTrash`
- Upload/Download when valid `from-to` flag provided (BlobPipe, PipeBlob) => Success
- Upload when invalid stdPipeIn provided => `fatal: failed to read from Stdin due to error` 
- Download when no stdPipeOut provided => Success. Echoed content on the terminal